code/modules/mob/living/carbon/human/human_movement.dm

```diff
@@ -1,91 +1,75 @@
-		if (bodytemperature < 183.222)
-			tally += (283.222 - bodytemperature) / 10 * 1.75
-	else if (undergoing_hypothermia())
-		tally += 2*undergoing_hypothermia()
-
-	//(/vg/ EDIT disabling for now) handle_embedded_objects() //Moving with objects stuck in you can cause bad times.
+			return min(..(), 1)
+	return ..()
 
-	if(reagents.has_reagent(NUKA_COLA))
-		tally -= 10
+/mob/living/carbon/human/base_movement_tally()
+	. = ..()
 
-	if((M_RUN in mutations))
-		tally -= 10
+	if(flying)
+		return // Calculate none of the following because we're technically on a vehicle
+	if(reagents.has_reagent(HYPERZINE))
+		return // Hyperzine ignores slowdown
+	if(istype(loc, /turf/space))
+		return // Space ignores slowdown
 
-	var/health_deficiency = (100 - health - halloss)
-	if(health_deficiency >= 40)
-		tally += (health_deficiency / 25)
+	if (species && species.move_speed_mod)
+		. += species.move_speed_mod
 
 	var/hungry = (500 - nutrition)/5 // So overeat would be 100 and default level would be 80
 	if (hungry >= 70)
-		tally += hungry/50
-
-	if(wear_suit)
-		tally += wear_suit.slowdown
-
-	if(shoes)
-		tally += shoes.slowdown
-
-	for(var/obj/item/I in held_items)
-		if(I.flags & SLOWDOWN_WHEN_CARRIED)
-			tally += I.slowdown
+		. += hungry/50
 
-	for(var/organ_name in list(LIMB_LEFT_FOOT,LIMB_RIGHT_FOOT,LIMB_LEFT_LEG,LIMB_RIGHT_LEG))
-		var/datum/organ/external/E = get_organ(organ_name)
-		if(!E || (E.status & ORGAN_DESTROYED))
-			tally += 4
-		if(E.status & ORGAN_SPLINTED)
-			tally += 0.5
-		else if(E.status & ORGAN_BROKEN)
-			tally += 1.5
-
-	if(pain_shock_stage >= 50)
-		tally += 3
-
-	if(M_FAT in src.mutations)
-		tally += 1.5
-
-	var/skate_bonus = 0
-	var/disease_slow = 0
-	for(var/obj/item/weapon/bomberman/dispenser in src)
-		disease_slow = max(disease_slow, dispenser.slow)
-		skate_bonus = max(skate_bonus, dispenser.speed_bonus)//if the player is carrying multiple BBD for some reason, he'll benefit from the speed bonus of the most upgraded one
-	tally = tally - skate_bonus + (6 * disease_slow)
+	if (dna.mutantrace == "slime")
+		if (bodytemperature < 183.222)
+			. += (283.222 - bodytemperature) / 10 * 175 // MAGIC NUMBERS!
+	else if (undergoing_hypothermia())
+		. += 2*undergoing_hypothermia()
+
+	if(feels_pain() && !has_painkillers())
+		if(pain_shock_stage >= 50)
+			. += 3
+		
+		for(var/organ_name in list(LIMB_LEFT_FOOT,LIMB_RIGHT_FOOT,LIMB_LEFT_LEG,LIMB_RIGHT_LEG))
+			var/datum/organ/external/E = get_organ(organ_name)
+			if(!E || (E.status & ORGAN_DESTROYED))
+				. += 4
+			if(E.status & ORGAN_SPLINTED)
+				. += 0.5
+			else if(E.status & ORGAN_BROKEN)
+				. += 1.5
+
+/mob/living/carbon/human/movement_tally_multiplier()
+	. = ..()
 
-	if(reagents.has_reagent(HYPERZINE))
+	if(!reagents.has_reagent(HYPERZINE))
+		if(!shoes)
+			. *= NO_SHOES_SLOWDOWN
+		if(M_FAT in mutations) // hyperzine can't save you, fatty!
+			. *= 1.5
+		if(M_RUN in mutations)
+			. *= 0.8
+
+		if(reagents.has_reagent(NUKA_COLA))
```

Comment:
Just to be sure, you're intentionally not allowing stacking NUKA_COLA and HYPERZINE anymore, right?